src/handlers/paDeficit.js

```diff
@@ -0,0 +1,145 @@
+const {
+  getEthersProvider,
+  Finding,
+  FindingSeverity,
+  FindingType,
+  ethers,
+} = require("forta-agent");
+
+const { WAD_DECIMALS } = require("../constants");
+
+const config = require("../config.json");
+
+const PREMIUMS_ACCOUNT_ABI =
+  '[{"inputs": [], "name": "activePurePremiums", "outputs": [ { "internalType": "uint256", "name": "", "type": "uint256" } ], "stateMutability": "view", "type": "function" }, { "inputs": [], "name": "surplus", "outputs": [ { "internalType": "int256", "name": "", "type": "int256" } ], "stateMutability": "view", "type": "function" }]';
+
+const AMOUNT_TO_WAD = ethers.BigNumber.from("10").pow(
+  ethers.BigNumber.from(WAD_DECIMALS - config.erc20Tokens.USDC.decimals)
+);
+
+const premiumsAccounts = config.handlers.paDeficit.premiumsAccounts;
+
+function createHandleBlock(
+  getEthersProvider,
+  premiumsAccounts,
+  paContractGetter
+) {
+  const provider = getEthersProvider();
+  const paContractFactory = (pa) => paContractGetter(pa, provider);
+
+  async function handleBlock(blockEvent) {
+    const findings = [];
+
+    await Promise.all(
+      premiumsAccounts.map(async (pa) => {
+        const contract = paContractFactory(pa);
+
+        const deficit = await getDeficit(contract, blockEvent.blockNumber);
+
+        const activePurePremiums = await getActivePurePremiums(
+          contract,
+          blockEvent.blockNumber
+        );
+
+        const ratio = deficit
+          .div(activePurePremiums)
+          .mul(
+            ethers.BigNumber.from("10").pow(config.erc20Tokens.USDC.decimals)
```

Comment:
Deberías tener una librería para operar entre valores wad, similar a https://github.com/ensuro/ensuro/blob/main/contracts/dependencies/WadRayMath.sol

Esta operación que hacés supongo que funciona porque estás operando un wad (18 decimales), dividido por un amount 6 decimales. Pero para evitar esa "magia" lo mejor sería que siempre conviertas los valores a wad y luego hagas operaciones `wadMul` y `wadDiv` y finalmente wad to float.

Wad mul, usando BigNumber (que creo que no tiene límite de 256bits como la versión online), sería hacer `(a*b/10**18)` y wadDiv sería `a*10**18/b` (creo...).

Otra alternativa (ahora que lo pienso me suena mejor) sería que las f convertir a float (o mejor a big.js, la misma librería que usamos en el front para decimales) y operar directamente en ese tipo. 

Las funciones getDeficit y getActivePurePremiums devolvería directamente un float o un decimal.